tools/cli/internal/openapi/oasdiff.go

```diff
@@ -29,12 +31,38 @@ type OasDiff struct {
 }
 
 func (o OasDiff) mergeSpecIntoBase() error {
-	return o.mergePaths()
+	if o.external == nil || o.external.Spec == nil {
+		return nil
+	}
+
+	if o.base == nil || o.base.Spec == nil {
+		*o.base = *o.external
```

Comment:
instead of pointer here make the receiver of the method a pointer, this is the preferred practice, also it's a code smell to mix pointer and non pointer receivers so they all should be pointers then